components/demo/demo-page.js

```diff
@@ -59,11 +66,22 @@ class DemoPage extends LitElement {
 		return html`
 			<main>
 				<h1 class="d2l-heading-2">${this.pageTitle}</h1>
-				<div class="d2l-demo-page-content"><slot></slot></div>
+				<div class="d2l-demo-page-content" @d2l-demo-snippet-fullscreen-toggle="${this._handleFullscreenToggle}"><slot></slot></div>
 			</main>
 		`;
 	}
 
+	async _handleFullscreenToggle() {
+		if (this.noScroll) {
+			this.noScroll = false;
+			await this.updateComplete;
+			document.documentElement.scrollTop = this._previousScrollTop;
+		}
+		else {
+			this._previousScrollTop = document.documentElement.scrollTop;
```

Comment:
Should `_previousScrollTop` be declared in some way?

Reply:
Unless it needs to trigger the render cycle to run for the component, it doesn't need to be declared as a Lit property. Having said that, sometimes we will initialize these types of member variables in the constructor, but I don't think we're consistent about that - initializing a member variable in the constructor to `undefined` is somewhat pointless, other than perhaps giving readers a little more insight as they dive into the code.